packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -100,7 +100,12 @@ class UpgradeCommandRunner {
     @required FlutterVersion flutterVersion,
     @required bool testFlow,
   }) async {
-    await verifyUpstreamConfigured();
+    final String upstreamRevision = await fetchRemoteRevision();
```

Comment:
I know we used to have this check, could you remind me of the reason we added? Is there git console spam on no-op upgrades otherwise?

Reply:
I think before, we never grabbed the revision that we wanted to upgrade to, so the way that we checked if an upgrade wasn't necessary was that we saved the version we were at, did an upgrade, and checked to see if we were on the same commit. if so, the upgrade was an essential no-op, and we didn't have to do the second half.

Since we're grabbing the desired upgrade target revision from the start, just exist early without messing with their repo.

Reply:
That makes sense, thanks Chris!